contrib/examples/apiserver/broker.yaml

```diff
@@ -8,6 +8,7 @@ spec:
   # service-catalog will use the contents of the secret. The secret should have "username"
   # and "password" keys
   authInfo:
-    basicAuthSecret:
+    secretRef:
```

Comment:
@arschles I prefer to keep `secretType` instead of duplicating the API and code, but it's up to the discussion of course.
Even if we were to eliminate the `secretType` property, I would still make a breaking change:
```yaml
authInfo:
    basicAuth:
        secret:
          namespace: some-namespace
          name: secret-name
```
and
```yaml
authInfo:
    bearerTokenAuth:
        secret:
          namespace: some-namespace
          name: secret-name
```
to decouple auth mode from secret.